### PR TITLE
Fix query and improve notifications tests

### PR DESCRIPTION
**Pull request by darioAnongba**
Fix query and improve notifications tests

### CODE REVIEW

File: tests/e2e/notifications_test.go [hunk 1]
```diff
@@ -156,3 +153,51 @@ func (s *notificationsTestSuite) TestSuspend() {
...
+
+			retries++
+			if retries >= 6 {
+				return nil, fmt.Errorf("status was not updated to %s", status)
+			}
+		}
+	}
+}
+
+func (s *notificationsTestSuite) waitForMined(jobUUID string) error {
```
Comment (6l20) [inline tests/e2e/notifications_test.go:181]: Here naming could be `waitForJobMined`

File: tests/e2e/notifications_test.go [hunk 2]
```diff
@@ -156,3 +153,51 @@ func (s *notificationsTestSuite) TestSuspend() {
 		require.Equal(t, tx2.UUID, receipt2.Data.(*entities.Job).ScheduleUUID)
 	})
 }
+
+func (s *notificationsTestSuite) waitForStatus(eventStreamUUID string, status entities.EventStreamStatus) (*types.EventStreamResponse, error) {
```
[the remark this answers is not in the document]
Comment (darioAnongba) [reply tests/e2e/notifications_test.go]: done

File: tests/e2e/notifications_test.go (hunk 1 above)
Comment (darioAnongba) [reply tests/e2e/notifications_test.go]: done

File: src/entities/notification.go [hunk 3]
```diff
@@ -38,16 +38,17 @@ func (n *NotificationSourceType) String() string {
...
-	Type            NotificationType
-	APIVersion      string
-	Job             *Job
-	EventLogs       []*ethereum.Log
-	CreatedAt       time.Time
-	UpdatedAt       time.Time
-	Error           string
+	JobUUID          string
+	SubscriptionUUID string
+	SourceType       NotificationSourceType
```
Comment (ggarri) [inline src/entities/notification.go:43]: Since `sourceUUID` disappears, does it makes sense to keep this?

**Review by ggarri — APPROVED** [review]

File: src/entities/notification.go (hunk 3 above)
Comment (darioAnongba) [reply src/entities/notification.go:43]: yes it's necessary to keep it to know which of the `JobUUID` or `SubscriptionUUID` should be defined.

**Pull request merged**